src/test/java/com/synopsys/integration/detect/tool/signaturescanner/PublishSignatureScanReportsTest.java

```diff
@@ -0,0 +1,150 @@
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import com.synopsys.integration.detect.configuration.enumeration.ExitCodeType;
+import com.synopsys.integration.detect.lifecycle.shutdown.ExitCodePublisher;
+import com.synopsys.integration.detect.lifecycle.shutdown.ExitCodeRequest;
+import com.synopsys.integration.detect.tool.signaturescanner.operation.PublishSignatureScanReports;
+import com.synopsys.integration.detect.workflow.status.DetectIssue;
+import com.synopsys.integration.detect.workflow.status.Status;
+import com.synopsys.integration.detect.workflow.status.StatusEventPublisher;
+import com.synopsys.integration.detect.workflow.status.StatusType;
+
+public class PublishSignatureScanReportsTest {
+    @Test
+    public void treatsSkippedAsSuccess() {
+        List<SignatureScannerReport> reports = Arrays.asList(
+            skippedReport("skipped", 2),
+            successfulReport("success", 1)
+        );
+
+        EventAssert publishedEvents = publishReports(reports, true);
+
+        publishedEvents.assertWinningExitCode(ExitCodeType.SUCCESS);
+    }
+
+    @Test
+    public void treatsSkippedAsIntervalNotMet() {
+        List<SignatureScannerReport> reports = Arrays.asList(
+            skippedReport("skipped", 2),
+            successfulReport("success", 1)
+        );
+
+        EventAssert publishedEvents = publishReports(reports, false);
+
+        publishedEvents.assertWinningExitCode(ExitCodeType.FAILURE_MINIMUM_INTERVAL_NOT_MET);
+    }
+
+    @Test
+    public void treatsStatusSummaryAsSuccess() {
+        List<SignatureScannerReport> reports = Collections.singletonList(
+            skippedReport("skipped", 2)
+        );
+
+        EventAssert publishedEvents = publishReports(reports, true);
+
+        publishedEvents.assertOnlySuccesfullStatusSummaries();
+        publishedEvents.assertNoIssues();
+    }
+
+    @Test
+    public void skippedIsAnIssueAndFailure() {
+        List<SignatureScannerReport> reports = Collections.singletonList(
+            skippedReport("skipped", 2)
+        );
+
+        EventAssert publishedEvents = publishReports(reports, false);
+
+        publishedEvents.assertOnlyFailureStatusSummaries();
+        publishedEvents.assertAnIssueContains("minimum scan interval");
+    }
+
+    public EventAssert publishReports(List<SignatureScannerReport> reports, boolean treatSkippedAsSuccess) {
+        InMemoryStatusEventPublisher statusEventPublisher = new InMemoryStatusEventPublisher();
+        InMemoryExitCodePublisher exitCodePublisher = new InMemoryExitCodePublisher();
+        PublishSignatureScanReports publishSignatureScanReports = new PublishSignatureScanReports(exitCodePublisher, statusEventPublisher, treatSkippedAsSuccess);
+        publishSignatureScanReports.publishReports(reports);
+
+        return new EventAssert(exitCodePublisher, statusEventPublisher);
+    }
+
+    private class EventAssert {
```

Comment:
This is neat and  I like this way this reads. Could this be used by similar tests?

Reply:
Yeah I didn't know if we wanted generic utils for events so I just put in here but as soon as another test wants it I will happily factor it out.